reports/report_details.py

```diff
@@ -61,822 +61,1099 @@ def add_transnational_to_regions(regions):
 
 WS_INFO = {
     'ws_01': {
-        'name': '1',
-        'title': 'Participating Countries',
-        'desc': 'Breakdown of all media by region',
-        'reports': ['global'],
-        'historical': '1F',
+        '2015': {
+            'name': '1',
+            'historical': '1F',
```

Comment:
The reason for keeping this is because that's the key that the data is stored in (by year) in `historical.json`. @kilemensi

Reply:
You must be thinking about something else @ascii-dev. 👇🏾 is the 2015 we're importing. There is no `1F`.
![Screenshot from 2020-12-14 18-18-04](https://user-images.githubusercontent.com/1779590/102099059-ed952f80-3e38-11eb-84fe-0627056220b0.png)

Again `historical` means which sheet is the value coming from for the **year** in question. In `2010`, yes, it's coming from `1F` and it's going to sheet `1` (i.e. `name`) in _2015 report_. In `2015`, it's coming  from `1` and it's going to `1` in _2020 report_.

Reply:
This is what I'm talking about:
![image](https://user-images.githubusercontent.com/8318974/102100266-891ea400-3e29-11eb-96c4-c18934b911de.png)

Reply:
@kilemensi

Reply:
At the moment, I pull in the data tagged `2015` in the `work sheets` and attach them to the current data in the `historical.json` file with the same data from the past years, simply by adding a new key for the current year to that same object. @kilemensi

Reply:
... we've gone over this long enough now @ascii-dev: **We haven't made 2020 changes yet so the current report structure is the same as it was  back in 2015**. In `2015`, yes all historical data needed to be in `1F`. In `2020` for the first time, historical data will be in `1`.

Reply:
I understand all that part. So, that means, when we start including `2020` data (which we haven't gotten to yet) we'll have to rewrite the whole `historical.json` file so that all `1F` data gets written to a `1` key, right? @kilemensi

Reply:
The current dashboard already has 2020 data @ascii-dev ... the report @esirK generated is using 2020 data but **2015 structure**. This PR is the start of introducing **2020 structure** and that's why we're doing it one sheet at a time & we're also preserving 2015 structure in case someone need to run the report while we're still working.

Otherwise, why would we bother with all these changes?

Reply:
So, if I get where we're at now, we need to have the `historical.json` file rewritten so that all `1F` data gets written to a `1` key? @kilemensi

Reply:
No, we don't need to do any such thing. My understanding:

1. historical.py loads historical.json when instantiated.
2. When you run import command, it basically goes through the Excel you provided and update the values if they already exist, or create new once if they don't exist.
3. The file is key based and the key is the year... For 2020, it will be only interested in 2015.
4. When running generate report, historical.json will be loaded then only those sheets that have 2015 data will be used if they align with 2020 structure... So it doesn't matter what's in 1F because 2020 doesn't have 1F so it won't even look what's there...
5. If however 1 doesn't have 2015, 2020 report will be generate 1 without historical data which is what we're fixing here: Aligning what 2020 expects from historical.

Reply:
The root keys are not year based, they are `1F` based. Will it be possible to jump on a call? @kilemensi

Reply:
Sorry went for a run... Yes, @ascii-dev , the end results is the same, it all depends on what we choose to use

Take https://github.com/CodeForAfrica/gmmp/blob/d84a3da0c8cf88ac4be8c0dcfec4b02c0056de9c/reports/report_details.py#L62-L69

It translates to:

> When creating GMMP 2015 report, worksheet 1 (name is 1) should load historical data from worksheet 1F of GMMP 2010 report.

Here the years are **_implied_** and **not** clearly **_stated_** since they basically only deal with only one report structure. In 2020 however, we're going to make the years **_explicit_** since we'll be supporting **two** report structures: GMMP 2020 and GMMP 2015.

One structure could be:

```python
  WS_INFO = {
    'ws_01': {
        '2010': {
            'name': '1',
            'historical': '1F',
           ...
        },
        '2015': {
             'name': '1',
            'historical': '1',
           ...
        },
   ....
```

Which translates to:

> When creating worksheet one of the report, if you're creating GMMP 2015 report, historical data should come from worksheet 1F of the 2010 GMMP report... and if you're creating GMMP 2020 report, historical data should come from worksheet 1 of the GMMP 2015 report.

The exact same thing could be said as 

```python
  WS_INFO = {
      '2010': {
        'ws_01': {
            'name': '1',
            'historical': '1F',
           ...
        },
        'ws_02': {
           ...
        },
      '2015: {
        'ws_01': {
            'name': '1',
            'historical': '11',
   ...
```

Which translates to:

> When creating GMMP 2015 report, historical data should come 2010 GMMP report. Worksheet 1 of GMMP 2015  should load data from worksheet 1F of the GMMP 2010 report. Likewise when creating GMMP 2020 report, historical data should come from the GMMP 2015 report. Worksheet 1 of 2020 report should load historical data from worksheet 1 of 2015 report.

Both of these could work.... One you deal with year in every worksheet; The other you deal with years at top level and forget about it.